crates/bevy_ecs/src/system/system_param.rs

```diff
@@ -1023,6 +1055,111 @@ unsafe impl<'w> SystemParam for DeferredWorld<'w> {
     ) -> Self::Item<'world, 'state> {
         world.into_deferred()
     }
+
+    fn is_exclusive() -> bool {
+        true
+    }
+}
+
+const MUT_WORLD_ERROR: &str = "&mut World requires exclusive access to the \
+    entire world, but a previous system parameter already registered access to \
+    a part of it. Allowing this would break Rust's mutability rules";
+
+// SAFETY: `&mut World` registers exclusive access to the entire world.
+unsafe impl SystemParam for &mut World {
+    type State = ();
+    type Item<'w, 's> = &'w mut World;
+
+    fn init_state(_world: &mut World, system_meta: &mut SystemMeta) -> Self::State {
+        let mut access = Access::default();
+        access.read_all();
+        if !system_meta
+            .archetype_component_access
+            .is_compatible(&access)
+        {
+            panic!("{}", MUT_WORLD_ERROR);
+        }
+        system_meta.archetype_component_access.extend(&access);
+
+        let mut filtered_access = FilteredAccess::default();
+
+        filtered_access.read_all();
```

Comment:
I feel like we shouldn't set any filtered_access as &mut World doesn't filter.

Reply:
> I feel like we shouldn't set any filtered_access as &mut World doesn't filter.

`component_access_set` is the one that actually gets checked for conflicts by things like `Res` and `Query`, so you do want to populate it.  `FilteredAccess::default()` is unfiltered, so this is declaring read access with no filter.  It could also be written `FilteredAccess::matches_everything()`, which may be more clear.